### DISCUSSION TITLE
Approach towards changing @Comfy-org/litegraph

### DISCUSSION THREAD

**Discussion by christian-byrne** [Comfy-Org/ComfyUI_frontend#114] [General]
I noticed [Comfy-Org/litegraph#e361b39](https://github.com/Comfy-Org/litegraph.js/commit/e361b39f364bfe17559043b14e7b469ed41bffcc). What is the approach toward changing @Comfy-org/litegraph?

There are some small issues which can be fixed relatively easily with non-breaking (AFAICT) changes to litegraph, but would require relatively complex solutions using hooks. Here are some examples:

- [truncate combo widget display text without changing actual value](https://github.com/comfyanonymous/ComfyUI/issues/3335)
- [number 6 from this issue - close prompt dialogs on click outside](https://github.com/comfyanonymous/ComfyUI/issues/90)
- [don't render prompt dialog offscreen](https://github.com/comfyanonymous/ComfyUI/issues/3155)
- [auto-select and focus input value when opening prompt dialog](https://github.com/comfyanonymous/ComfyUI/issues/1023)
- [on prompt dialogs, changing the HTMLInputElement.type attribute to number when it's a number](https://github.com/comfyanonymous/ComfyUI/issues/1612)
- [add shortcut to change node title](https://github.com/comfyanonymous/ComfyUI/issues/557)

**Comment by huchenlei** [comment]
I think the general approach now is to make whatever changes necessary to litegraph and make sure these changes are tested. You can refer to https://github.com/Comfy-Org/ComfyUI_frontend/commit/525adb754f9499ef48400b83b6bfe76eb64e7321 that added tests on the new events.

There are a couple of other litegraph rewrites floating around as well. So we want changes well documented and tested, just in case if we want to switch to another litegraph impl later.
